### PR TITLE
Fix search input for latest version of Algolia

### LINKED ISSUE #738: Algolia Search: DocSearch API v3 doesn't seem to have 'inputSelector' any more
Hi there, I'm trying to add Algolia Docsearch to www.graphviz.org.

The Docsy docs at https://www.docsy.dev/docs/adding-content/navigation/#adding-algolia-docsearch say

> Update the `inputSelector` field in the body end Javascript with the appropriate CSS selector (e.g. `.td-search-input` to use the default CSS from this theme).

I can't find an `inputSelector` option in the Docsearch API v3: https://docsearch.algolia.com/docs/api. It looks like this was a thing for Docsearch API v2: https://docsearch.algolia.com/docs/legacy/dropdown.

The DocSearch API v3 wants you to give a container that they render the searchbox into, it seems.

I'll probably just roll with API v2 for now (so this isn't an urgent issue or anything), just wanted to file it as FYI.

### LINKED ISSUE #845: New version of Algolia search breaks DOCSY search
There is a new version of Algolia search that has a different interface that doesn't seem to work with Docsy.

The new Algolia DocSearch no longer uses the ".td-search-input" as an input parameter in the new javascript.
https://www.docsy.dev/docs/adding-content/navigation/#configure-algolia-docsearch

The new v3 Algolia JS doco is here and uses "container" instead
https://docsearch.algolia.com/docs/docsearch-v3/

This seems like something Docsy has to update to?  Not sure what the implications are here, but posting here for someone who might know a bit more about this.

Thanks Docsy team!

### PR DESCRIPTION
**Pull request by LisaFC**
- Fixes #845
- Closes #738

Preview: https://deploy-preview-985--docsydocs.netlify.app/docs/adding-content/navigation/#configure-algolia-docsearch

WIP, need to make sure docs instructions also apply to latest version.

### CODE REVIEW

**Comment by LisaFC** [comment]
Docs now done!

**Comment by LisaFC** [comment]
I've marked this as breaking as it will not work with Algolia DocSearch v2 - on the other hand, per #845 the old version of Algolia DocSearch doesn't work any more (v2 indexes are no longer being updated) so I'm not sure if it will make v2 users any more broken....

@chalin @geriom @emckean ideas for publicizing this? I presume we will have some v2 users who possibly haven't noticed their new content hasn't been indexed. Migration from v2 should be pretty trivial, I think the index name and search API key are the same, just the format of the snippet is different: https://docsearch.algolia.com/docs/migrating-from-v2

**Comment by LisaFC** [comment]
@chalin ready for review!

File: layouts/partials/search-input.html [hunk 1]
```diff
@@ -1,5 +1,7 @@
-{{ if or .Site.Params.gcs_engine_id .Site.Params.algolia_docsearch -}}
+{{ if .Site.Params.gcs_engine_id -}}
 <input type="search" class="form-control td-search-input" placeholder="&#xf002; {{ T "ui_search" }}" aria-label="{{ T "ui_search" }}" autocomplete="off">
+{{ else if .Site.Params.algolia_docsearch -}}
+<div id="docsearch"/>
```
Comment (chalin) [inline layouts/partials/search-input.html:4]: Hmm, I don't think that a `div` can be a self-closing tag. And ... SO seems to agree: https://stackoverflow.com/q/7971716/3046255

**Review by chalin — CHANGES_REQUESTED** [review]
Thanks for the update. See inline comments.

Have you tested out the instructions?

File: userguide/content/en/docs/Adding content/navigation.md [hunk 2]
```diff
@@ -210,11 +214,30 @@ If you are accepted to the program, you will receive the JavaScript code to add
...
 
-3. Add the JavaScript code provided to you by Algolia to the head and body of every page on your site. See [Add code to head or before body end](/docs/adding-content/lookandfeel/#add-code-to-head-or-before-body-end) for details.
+3. Add the CSS and JS to use Algolia to the head and body of every page in your site, following the instructions in [Add code to head or before body end](/docs/adding-content/lookandfeel/#add-code-to-head-or-before-body-end).
+   
+   * In `head-end.html` add the DocSearch CSS:
+
+      ```html
+      <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css@3" />
+      ```
+
```
Comment (chalin) [inline userguide/content/en/docs/Adding content/navigation.md:224]: Come to think of it, why not just move this into `head-end.html`, relieving projects from doing one manual step?

**Comment by chalin** [comment]
@LisaFC - Oh, just remembered: as we discussed during the PSC meeting, could you add an entry to the CHANGELOG?

File: userguide/content/en/docs/Adding content/navigation.md [hunk 3]
```diff
@@ -189,13 +189,17 @@ If you don't specify a Google Custom Search Engine ID for your project and haven
 
 ## Configure Algolia DocSearch
 
-As an alternative to GCSE, you can use [Algolia DocSearch](https://docsearch.algolia.com/) with this theme. Algolia DocSearch is free for public documentation sites.
+As an alternative to GCSE, you can use [Algolia DocSearch](https://docsearch.algolia.com/) with this theme. Algolia DocSearch is free for public documentation sites.  Docsy supports Algolia DocSearch V3.
+
+{{% alert title="Note" %}}
+Docsy previously supported Algolia DocSearch V2, which is now deprecated. If you are an existing Algolia DocSearch V2 user and want to use the latest Docsy version, [follow the migration instructions](https://docsearch.algolia.com/docs/migrating-from-v2) in the DocSearch documentation to update your DocSearch code snippet.
+{{% /alert %}}
```
[the remark this answers is not in the document]
Comment (LisaFC) [reply userguide/content/en/docs/Adding content/navigation.md]: Done.

File: layouts/partials/search-input.html (hunk 1 above)
Comment (LisaFC) [reply layouts/partials/search-input.html]: Done.

**Comment by LisaFC** [comment]
Yes, instructions tested, I added DocSearch to a test site here (using the Algolia site index): https://tender-beaver-c94732.netlify.app/

**Review by chalin — APPROVED** [review]
LGTM

Will you be adding to the CHANGELOG file via this or a separate PR?

File: userguide/content/en/docs/Adding content/navigation.md (hunk 2 above)
Comment (LisaFC) [reply userguide/content/en/docs/Adding content/navigation.md:224]: Hmm, I did it this way as that's how we did it in the previous integration, it's always been basically an empty file - also not sure if users would want a random CSS include on all their pages if they're not using Algolia. Thoughts?

**Comment by LisaFC** [comment]
I can add it in this PR - what format do we want, should we put it under a new 0.3.0 release?

**Comment by chalin** [comment]
> put it under a new 0.3.0 release?

Yes.

File: userguide/content/en/docs/Adding content/navigation.md (hunk 2 above)
Comment (chalin) [reply userguide/content/en/docs/Adding content/navigation.md:224]: Maybe the file _should_ be empty, but it currently isn't: it contains logic that tests whether Algolia is configured. (We can address this through a separate PR.)

Comment (LisaFC) [reply userguide/content/en/docs/Adding content/navigation.md:224]: Gotcha.

**Review by chalin — APPROVED** [review]

**Comment by chalin** [comment]
> Should I merge this now or do we need to do more release-y things?

Thanks for your patience. I wanted to adjust the CHANGELOG a bit. That's been done.

**Pull request merged**